File: test/nodejs/rfuns_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`rfuns extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.RFUNS_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

});

File: test/python/rfuns_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the rfuns extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('RFUNS_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `RFUNS_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn
